rllib/algorithms/sac/sac.py

```diff
@@ -371,3 +430,133 @@ def get_default_policy_class(
+                # list of Episodes.
+                # TODO (simon): Change this to a preprocessing in the learner.
+                # postprocessed_episodes = self.postprocess_episodes(episodes)
+
+            episodes = tree.flatten(episodes)
+            # TODO (sven): single- vs multi-agent.
+            self._counters[NUM_AGENT_STEPS_SAMPLED] += sum(len(e) for e in episodes)
+            self._counters[NUM_ENV_STEPS_SAMPLED] += sum(len(e) for e in episodes)
+
+            # Add the sampled experiences to the replay buffer.
+            # TODO (simon): Apply the same function like in PPO to aggregate episodes
+            # over workers.
+            self.local_replay_buffer.add(episodes)
+
+        # Update the target network each `target_network_update_freq` steps.
+        current_ts = self._counters[
+            NUM_AGENT_STEPS_SAMPLED
+            if self.config.count_steps_by == "agent_steps"
+            else NUM_ENV_STEPS_SAMPLED
+        ]
+
+        # If enough experiences have been sampled start training.
+        if current_ts > self.config.num_steps_sampled_before_learning_starts:
+            # Run multiple training iterations.
+            for _ in range(sample_and_train_weight):
+                # Sample training batch from replay_buffer.
+                train_dict = self.local_replay_buffer.sample(
+                    num_items=self.config.train_batch_size,
+                    # TODO (simon): Implement for n-step adjustment.
+                    # batch_length_T=self.config.n_step + 1,
+                    batch_length_T=2,
+                )
+
+                # TODO (sven): This looks a bit ugly, but we have not the same
+                # naming in `SingleAgentEpisode` and `SampleBatch`.
+                # Note, the `SingleAgentEpisode` stores obs_t. action_(t-1),
+                # reward_t, infos_t, while the replay buffer's `sample()`
+                # returns obs_t, action_t, reward_(t-1). So, we take the
+                # reward from the subsequent experience in the episode to
+                # calculate the Q-values.
+                train_batch = SampleBatch(
+                    {
+                        SampleBatch.OBS: train_dict[SampleBatch.OBS][:, 0],
+                        SampleBatch.NEXT_OBS: train_dict[SampleBatch.OBS][:, 1],
+                        SampleBatch.ACTIONS: train_dict[SampleBatch.ACTIONS][:, 0],
+                        SampleBatch.REWARDS: train_dict[SampleBatch.REWARDS][:, 1],
+                        SampleBatch.TERMINATEDS: train_dict["is_terminated"][:, 0],
+                        SampleBatch.TRUNCATEDS: train_dict["is_truncated"][:, 0],
+                        "is_first": train_dict["is_first"][:, 0],
+                        "is_last": train_dict["is_last"][:, 0],
+                    }
+                )
+                train_batch = train_batch.as_multi_agent()
+
+                # Training on batch.
+                train_results = self.learner_group.update(
```

Comment:
New API:
`LearnerGroup.update_from_batch()`

Also, leave `minibatch_size` and `num_iters` out here (default takes care of these in this case).

Reply:
Done